mvcc/kv.go

```diff
@@ -111,6 +111,9 @@ type KV interface {
 	// This method is designed for consistency checking purposes.
 	Hash() (hash uint32, revision int64, err error)
 
+	// HashByRev retrieves the hash of KV state given a revision.
+	HashByRev(rev int64) (hash uint32, compactRev int64, err error)
```

Comment:
Hash includes all state (including leases for example). HashByRev only hashes the kv state for good reason (lease does not understand rev yet...). we need to make this clear though.